### DISCUSSION TITLE
API for managing secrets - extracted from match

### DISCUSSION THREAD

**Discussion by lacostej** [fastlane/fastlane#18230] [Ideas]
match is very useful for iOS support. We use @hjanuschka's cryptex for everything else. I think cryptex was originally forked from match. And now both have slightly divergedd, especially with match supporting various storage backends now. And it looks like there haven't been too many commits on `cryptex` lately so I don't expect it to catch up anymore.

I see in issues & discussions people asking for a more generic tool to store secrets. Is there something out there that I am missing?

If not, I think it could be cryptex. But maybe one idea could be to extract the part of match that deals with encryption and storage, and reuse that to upgrade it. Or to use it as baseline to create a new action.

Thoughts?

See also potentially related threads
* #18215

**Comment by dinsen** [comment]
Hi @lacostej 
I like the idea of moving the storage part out of match and make a more generic tool to store secrets. 

I'm just not sure how we should handle this going forward as I haven't been around for that long.
